pandas/_libs/parsers.pyx

```diff
@@ -1752,18 +1766,40 @@ cdef inline int _try_double_nogil(parser_t *parser,
         char *p_end
         khiter_t k, k64
 
-    na_count[0] = 0
-    coliter_setup(&it, parser, col, line_start)
+    with nogil(double_converter_t is double_converter_with_nogil_t):
```

Comment:
the motivating use case for conditional nogil was `with nogil(something_t is not object)`; is that pattern not available here?  (this file is one im less familiar with)

Reply:
I'm not sure I understand your question. 

- Conditional gil is resolved at compile-time. 
- The use of the double_converter_t fused type generates two versions of this function. 
- This new `nogil(condition)` idiom is used to take the gil or not based on the specific type variant. 
- The condition is true for one generated version of the function and false for the other.